src/phoenix/server/app.py

```diff
@@ -302,8 +302,48 @@ async def api_docs(request: Request) -> Response:
     return get_swagger_ui_html(openapi_url="/schema", title="arize-phoenix API")
 
 
-def create_app(
+class SessionFactory:
+    def __init__(
+        self,
+        session_factory: Callable[[], AsyncContextManager[AsyncSession]],
+        dialect: str,
+        engine: Optional[AsyncEngine] = None,
```

Comment:
Why not make it more strict?
```suggestion
        dialect: SupportedSQLDialect,
        engine: AsyncEngine,
```

Reply:
great idea